test/helper.js

```diff
@@ -7,6 +7,40 @@ const pid = process.pid
 const hostname = os.hostname()
 const v = 1
 
+function expected (str) {
+  var memo = new WeakMap()
+
+  var valid = {
+    string: n => n !== void 0 && n.length > 0 && typeof n === 'string',
+    array: n => Array.isArray(n) && n.length > 0,
+    object: n => n !== null && typeof n === 'object'
+  }
+  var cursor = {
+    set: (k, v) => (valid.object(k)) ? memo.set(k, v) : void 0,
+    get: (k) => (valid.object(k)) ? memo.get(k) : void 0,
+    has: (k) => (valid.object(k)) ? memo.has(k) : void 0
+  }
+  var shake = ({ s_, ndl_ }) => {
+    var delim = ' '
+    /* eslint-disable */
+    var sane = /([\s()[\[^\]]:*)/mg
+    /* eslint-enable */
+    if (!valid.string(s_) || !valid.array(ndl_)) return void 0
+    var tree = s_.replace(sane, ' ').split(delim).filter(o => ndl_.indexOf(o) !== -1)
+    return tree.length !== 0
+  }
+
+  return {
+    output: function output$ ({ has }) {
+      if (!valid.string(str)) return void 0
+      var prev = { str: str, has: has }
+      var tree = shake({ s_: str, ndl_: has })
+      if (!cursor.has(prev)) cursor.set(prev, tree)
+      return cursor.get(prev)
+    }
+  }
+}
```

Comment:
Can you please add some docs about this?

Reply:
What conventions for inline documentation is in project?

Reply:
Minor documentation added inline to clarify the purpose and usage